### PR TITLE
Use `….kernel.execute()` and change data contract

### PR DESCRIPTION
**Pull request by anatoly-eto**
- [x] Changed data contract.
  - Support for nested annotations (e.g., in `coco`) is still to be added.
- [x] Added a way to get results through `….notebook.kernel.execute()`.
  - `df.to_json(None, orient='records')` command is executed.

---

### Using `….kernel.execute()`:
- 
- 

---

### `coco` is still not supported:
-

### CODE REVIEW

File: src/sql/index.html [hunk 1]
```diff
@@ -28407,11 +28408,39 @@
           const [e, t] = x.exports.useState();
           return (
             x.exports.useEffect(() => {
-              fetch("./df.json")
-                .then((n) => n.json())
-                .then((n) => {
-                  t(n);
-                });
+              window.IPython ||
+                fetch("./oxford_pet.json")
```
Comment (changhiskhan) [inline src/sql/index.html:28412]: this is hard coded still

File: src/sql/index.html [hunk 2]
```diff
@@ -28407,11 +28408,39 @@
...
+                  .then((n) => {
+                    t(n);
+                  });
+            }, []),
+            x.exports.useEffect(() => {
+              (async () => {
+                if (!window.IPython) return;
+                console.log("executing\u2026"),
+                  await window.IPython.notebook.kernel.execute(
+                    "df.to_json(None, orient='records')",
```
Comment (changhiskhan) [inline src/sql/index.html:28423]: the variable `df` isn't always going to be the one you want.

**Review by changhiskhan — COMMENTED** [review]
there's still a hard-coded oxford_pet.json

also, `df` is not the right output variable to use. I think we need to generate a unique var name and use that instead

File: src/sql/index.html (hunk 1 above)
Comment (anatoly-eto) [reply src/sql/index.html:28412]: It is fallback for local development.
It never gets executed in notebook.

File: src/sql/index.html (hunk 2 above)
Comment (anatoly-eto) [reply src/sql/index.html:28423]: Yes-yes, we can even abstract the command to execute behind our private Python method.

**Comment by anatoly-eto** [comment]
Closing because widget was already updated here:
- https://github.com/eto-ai/ipython-sql/pull/4

**Pull request closed without merging**